src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java

```diff
@@ -39,15 +39,15 @@ public ResponseEntity generateFiling(@PathParam("transactionId") String transact
 
         if (transaction == null) {
             logRequestError(request, "no transaction in request session");
-            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```

Comment:
We can assume that the data is in a valid state (due to the interceptors) by the time it hits this controller, so if any of these objects are returning null, this should be an `INTERNAL_SERVER_ERROR` rather than a `BAD_REQUEST`. This applies to all of the changes you made regarding `BAD_REQUEST`.